src/sip/request.c

```diff
@@ -737,7 +737,10 @@ static int sip_request_send(struct sip_request *req, struct sip *sip,
 			return err;
 		}
 	}
-	else if (route->port) {
+	else if (route->port ||
+			(req->tp_selected &&
+				!dnsc_has_srv(req->sip->dnsc) &&
+				dnsc_getaddrinfo_enabled(req->sip->dnsc))) {
```

Comment:
This is hard to read and should be simplified (maybe a extra variable or if check).

Reply:
Maybe a `dnsc_gettadrinfo_only` function would be better here.

Reply:
Added `dnsc_gettadrinfo_only`.